### PR TITLE
fix(react-motion): sync internal value of canRender with presence

### PR DESCRIPTION
**Pull request by marcosmoura**
## Previous Behavior
`presence` values would only sync with `canRender` after a first render.

## New Behavior
`presence` and `canRender` is now always synced

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Avatar-mount_base.html">624</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Avatar-mount.html">617</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Button-mount_base.html">326</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Button-mount.html">316</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Field-mount_base.html">1086</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Field-mount.html">1089</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProvider-mount_base.html">697</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProvider-mount.html">680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-mount.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/MakeStyles-mount_base.html">862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/MakeStyles-mount.html">840</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Persona-mount_base.html">1705</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/Persona-mount.html">1653</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/SpinButton-mount_base.html">1342</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29394/perf-test-react-components/SpinButton-mount.html">1365</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `68.748 kB`<br />`19.742 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `206.308 kB`<br />`59.248 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `40.866 kB`<br />`13.546 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |     `6.503 kB`<br />`2.22 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e27181b1511f48817de41bffe4a1b4bb7941464d'>e27181b1511f48817de41bffe4a1b4bb7941464d</a></sub>
<div id="bundleSizeComment9423"/>

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-motion-preview/src/hooks/useMotion.ts [hunk 1]
```diff
@@ -177,11 +177,11 @@ function useMotionPresence<Element extends HTMLElement>(
       ref,
       type,
       active,
-      canRender: type !== 'unmounted',
+      canRender: presence || type !== 'unmounted',
```
Comment (Hotell) [inline packages/react-components/react-motion-preview/src/hooks/useMotion.ts:180]: can we add test that demonstrates behaviour / the fix respectively ?  from a person not involved I can only guess what is this about. ty

**Review by Hotell — APPROVED** [review]
pls see my comment

**Review by ValentinaKozlova — APPROVED** [review]

**Pull request merged**